internal/kafka/command_topic_produce.go

```diff
@@ -132,20 +132,21 @@ func (c *command) produce(cmd *cobra.Command, args []string) error {
 	signal.Notify(signals, os.Interrupt)
 	go func() {
 		<-signals
-		select {
-		case <-input:
-		default:
-			close(input)
-		}
+		input <- EOF
 	}()
 	// Prime reader
 	go scan()
 
 	deliveryChan := make(chan ckafka.Event)
 	for data := range input {
 		if data == "" {
```

Comment:
`switch data`?

Reply:
I don't think we can `switch data` because we need to `break` out of the loop over `input`.